### PR TITLE
CV evaluation for lying children added

### PR DESCRIPTION
**Pull request by lvonasek**
# Description

This PR adds CV evaluation for lying children. It uses ML segmentation to determine child pixels and uses highest, lowest 3D points to estimate height. The results are slightly worse than our ML solution. The value of this pull request is in having CV evaluation and height estimation for both lying and standing children.

The problematic of lying children is that the child pose is in most of the cases valid and we do not have solution yet to rate the body pose.

# User story
https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/1896

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests
- [x] New and existing unit tests pass locally with my changes
- [x] My Code is review by 1 People

### CODE REVIEW

File: cgmml/common/evaluation/CV/evaluation.py [hunk 1]
```diff
@@ -43,7 +43,11 @@ def filter_metadata(indata: list) -> list:
             continue
 
         # Check if it is a standing child
-        if not data[METADATA_SCAN_TYPE].startswith('10'):
+        if standing & (not data[METADATA_SCAN_TYPE].startswith('10')):
```
Comment (markus-hinsche) [inline cgmml/common/evaluation/CV/evaluation.py:46]:
```suggestion
        if standing and (not data[METADATA_SCAN_TYPE].startswith('10')):
```

File: cgmml/common/evaluation/CV/evaluation.py [hunk 2]
```diff
@@ -43,7 +43,11 @@ def filter_metadata(indata: list) -> list:
...
+            continue
+
+        # Check if it is a lying child
+        if (not standing) & (not data[METADATA_SCAN_TYPE].startswith('20')):
```
Comment (markus-hinsche) [inline cgmml/common/evaluation/CV/evaluation.py:50]:
```suggestion
        if (not standing) and (not data[METADATA_SCAN_TYPE].startswith('20')):
```

File: cgmml/common/evaluation/CV/height_prediction_with_ml_segmentation_lying.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+from cgmml.common.depthmap_toolkit.depthmap import Depthmap, is_google_tango_resolution
+from cgmml.common.depthmap_toolkit.depthmap_utils import calculate_boundary, vector_length
+from cgmml.common.background_segmentation.deeplab.deeplab import label_to_color_image
+from cgmml.common.background_segmentation.deeplab.deeplab_model import get_deeplab_model, PERSON_SEGMENTATION
+
+DEEPLAB_MODEL = get_deeplab_model()
+
+
+def predict_height(depthmap_file: str, rgb_file: str, calibration_file: str) -> Tuple[float, float]:
```
Comment (markus-hinsche) [inline cgmml/common/evaluation/CV/height_prediction_with_ml_segmentation_lying.py:14]: It seems like this duplicates a lot of code that we already have in `height_prediction_with_ml_segmentation.py`. 
Instead we could extract common functionality to a helper module which can be used in multiple places.

File: cgmml/common/evaluation/CV/evaluation.py [hunk 4]
```diff
@@ -32,7 +32,7 @@
 METADATA_ANGLE = 14
 
 
-def filter_metadata(indata: list) -> list:
+def filter_metadata(indata: list, standing: bool) -> list:
```
Comment (markus-hinsche) [inline cgmml/common/evaluation/CV/evaluation.py:35]:
```suggestion
def filter_metadata(indata: list, is_standing: bool) -> list:
```

https://www.comp.nus.edu.sg/~cs2103/AY1617S1/contents/coding-standards-java.html

File: cgmml/common/evaluation/CV/evaluation.py [hunk 5]
```diff
@@ -156,13 +161,17 @@ def process_height_prediction(depthmap_file: str, rgb_file: str, calibration_fil
     if len(sys.argv) != 4:
         print('You did not enter raw data path, metadata file name or method name')
         print('E.g.: python evaluation.py rawdata_dir metadata_file depthmap_toolkit')
-        print('Available methods are depthmap_toolkit and ml_segmentation')
+        print('Available methods are depthmap_toolkit, ml_segmentation and ml_segmentation_lying')
         sys.exit(1)
 
+    standing = True
```
Comment (markus-hinsche) [inline cgmml/common/evaluation/CV/evaluation.py:167]:
```suggestion
    is_standing = True
```

**Review by markus-hinsche — COMMENTED** [review]
Very good to have height estimation for lying children in the codebase!

Added a few coding suggestions

File: cgmml/common/depthmap_toolkit/depthmap.py [hunk 6]
```diff
@@ -484,6 +484,19 @@ def get_highest_point(self, mask: np.ndarray) -> np.ndarray:
...
+
+        scale_x = float(new_width) / float(self.width)
+        scale_y = float(new_height) / float(self.height)
+        self.cx = (self.cx - float(self.width) * 0.5) * scale_x + float(new_width) * 0.5
+        self.cy = (self.cy - float(self.height) * 0.5) * scale_y + float(new_height) * 0.5
+        self.fx *= scale_x
+        self.fy *= scale_y
+        self.width = int(new_width)
+        self.height = int(new_height)
+        self.depthmap_arr = np.zeros((self.width, self.height))
```
Comment (markus-hinsche) [inline cgmml/common/depthmap_toolkit/depthmap.py:498]: This function is intended to resize a depthmap. However this line seems to set the depthmap to zero.
Can you explain why?

Comment (lvonasek) [reply cgmml/common/depthmap_toolkit/depthmap.py:498]: Resizing is not rescaling. AFAIK we do not have an algorithm yet to do a proper rescaling of depthmap

**Review by markus-hinsche — APPROVED** [review]
LGTM

**Pull request merged**